File: app/app/Model/Member/Request/MemberUpdateProfileRequest.swift
struct MemberUpdateProfileRequest: Encodable {

    let publicImages: [ProfileImageUpdate]
    let privateImages: [ProfileImageUpdate]
    let nickname: String
    let birthYear: Int
    let bio: String?
}

struct ProfileImageUpdate: Encodable {

    let imageId: Int64?
    let key: String?
    let sortOrder: Int
}

File: server/src/main/kotlin/com/pidulgi/server/member/dto/request/MemberUpdateProfileRequest.kt
package com.pidulgi.server.member.dto.request

data class MemberUpdateProfileRequest(

    val images: List<ProfileImageUpdate> = emptyList(),
    val nickname: String,
    val birthYear: Int,
    val bio: String? = null,
) {

    data class ProfileImageUpdate(

        val imageId: Long? = null,
        val key: String? = null,
        val sortOrder: Int,
    )
}
